Python/Product/PythonTools/PythonToolsService.cs

```diff
@@ -231,7 +231,7 @@ public sealed class PythonToolsService : IDisposable {
         public VsProjectAnalyzer DefaultAnalyzer {
             get {
                 if (_analyzer == null) {
-                    _analyzer = _container.GetUIThread().Invoke(() => CreateAnalyzer());
+                    _analyzer = _container.GetUIThread().InvokeTask(() => CreateAnalyzerAsync()).WaitAndUnwrapExceptions();
```

Comment:
Could we not have a UIThread wrapper for JoinableTaskFactory.Run ([https://msdn.microsoft.com/en-us/library/dn283807.aspx](https://msdn.microsoft.com/en-us/library/dn283807.aspx)), that would take care of the blocking and not have to call WaitAndUnwrapExceptions here (and everywhere else we need something similar)?

Reply:
We do, but I added it after updating this call and didn't go back. (There are plenty of cases where we need to wait for an async task but don't have the UI thread available, so unfortunately we don't get a complete replacement, but in places where we can use the blocking function we should.)